pkg/protocols_test.go

```diff
@@ -59,19 +59,30 @@ func TestHttpProbe(t *testing.T) {
 // Creates an HTTP server for testing.
 func newTestHTTPServer(t *testing.T) *http.Server {
 	hostPort := net.JoinHostPort("127.0.0.1", "8080")
-	server := &http.Server{Addr: hostPort}
-	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+	mux := http.NewServeMux()
+	server := &http.Server{
+		Addr: hostPort,
+		Handler:mux,
+	}
+
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
 		io.WriteString(w, "pong\n")
 	})
+
+	l, err := net.Listen("tcp", hostPort)
+	if err != nil {
+		t.Fatalf("create listener %v", err)
+	}
+
 	go func() {
-		err := server.ListenAndServe()
-		if err != nil && err != http.ErrServerClosed {
+		if err := server.Serve(l); err != nil && err != http.ErrServerClosed {
```

Comment:
Use two different lines. Also to keep the formatting consistent with the codebase.